src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -137,6 +137,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DocumentationFile />
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(PublishDocumentationFile)' == ''">
+    <PublishDocumentationFile Condition="'$(GenerateDocumentationFile)' == 'true'">true</PublishDocumentationFile>
```

Comment:
> The situation currently is:  \[...\]

I see. In that case, I think we want this new opt-out to disable that behavior for project refs too.